doc/sphinx/user-extensions/syntax-extensions.rst

```diff
@@ -1508,21 +1509,163 @@ Numeral notations
          function application, constructors, inductive type families,
          sorts, and primitive integers) will be considered for printing.
 
-      :n:`( warning after @numeral )`
+         .. note::
+            When :n:`@qualid` is a parameterized inductive type, all
+            parameters must be provided via a notation.
+
+            .. coqtop:: in reset
+
+               Definition of_uint (u : Numeral.uint) : list unit :=
+                 let fix f n :=
+                   match n with O => nil | S n => cons tt (f n) end in
+                 f (Nat.of_num_uint u).
+               Definition to_uint (l : list unit) : Numeral.uint :=
+                 let fix f n :=
+                   match n with nil => O | cons tt l => S (f l) end in
+                 Nat.to_num_uint (f l).
+
+               Notation listunit := (list unit) (only parsing).
```

Comment:
This seems to work fine, i.e., both checks output `0 : list unit`.
Makes sense since the abbreviation is required only when defining the notation (the `Numeral Notation` line), not when the notation is used.